spec/lang/step/operators.md

```diff
@@ -346,14 +346,15 @@ impl<M: Memory> Machine<M> {
         };
         (Value::Bool(b), Type::Bool)
     }
-    /// Prepare a value for being compared with a `RelOp`. This involves turning it into an `Int`.
-    fn prepare_for_rel_op(val: Value<M>) -> Int {
+    /// Prepare a value for being compared with a `RelOp`. This involves turning it into a pair of `Int`s.
+    /// This pair is compared lexicographically, and is only important for wide pointers.
+    fn prepare_for_rel_op(val: Value<M>) -> (Int, Int) {
         match val {
-            Value::Int(i) => i,
-            Value::Bool(b) => if b { Int::from(1) } else { Int::from(0) },
-            Value::Ptr(p) if p.metadata.is_none() => p.thin_pointer.addr,
-            // TODO(UnsizedTypes): Prepare wide pointer for rel op: `addr * (MAX_META_VALUE+1) + meta_value`
-            Value::Ptr(_) => unimplemented!("comparing pointer with metadata"),
+            Value::Int(i) => (i, Int::ZERO),
```

Comment:
I think this is a bit too hacky... we shouldn't compare regular integers via pairs. Also the lexocographic part should be explicit in our sources, rather than relying implicitly on the order of `(Int, Int)`.

Instead maybe rename `eval_rel_op` to `eval_rel_op_int`, and then use that in `eval_bin_op` n a type-dependent way.

Also please match on `Value::Ptr` only once and then have a nested match on `metadata`.